### PR TITLE
refactor: refactor through model name with hasMany relationships

### PR DESCRIPTION
**Pull request by GuillaumeCisco**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Create automatic tests
- [x] No automatic tests failures
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

According to https://sequelize.org/master/manual/advanced-many-to-many.html
```
By passing a string to through above, we are asking Sequelize to automatically generate a model named User_Profiles as the through table (also known as junction table), with only two columns: userId and profileId. A composite unique key will be established on these two columns.
```

### CODE REVIEW

**Comment by arnaudbesnier** [comment]
Task linked: [CU-52jbcp P3 - Mysql model definition - createdAt field present in schema when timestamp is defined as false](https://app.clickup.com/t/52jbcp)

**Comment by closingin** [comment]
Duplicate of #445 😕

**Comment by GuillaumeCisco** [comment]
I've just reopened it for using the `through` key instead of modifying the `junctionTable` in the middle of a process.
It allows to be less error prone as we are more declarative.

We could remove the `junctionTable` key from the `reference` object if it does not break compatibility.

File: services/analyzer/sequelize-tables-analyzer.js [hunk 1]
```diff
@@ -139,6 +140,9 @@ function createReference(tableName, association, foreignKey, manyToManyForeignKe
     reference.ref = manyToManyForeignKey.foreignTableName;
     reference.otherKey = manyToManyForeignKey.columnName;
     reference.junctionTable = foreignKey.tableName;
```
Comment (closingin) [inline services/analyzer/sequelize-tables-analyzer.js:142]: Remove this variable if it's not used anywhere else

File: test-expected/sequelize/db-analysis-output/users.expected.json [hunk 2]
```diff
@@ -30,7 +30,8 @@
         "foreignKey": "user_id",
         "foreignKeyName": "userIdKey",
         "junctionTable": "user_books",
```
Comment (closingin) [inline test-expected/sequelize/db-analysis-output/users.expected.json:32]: Remove `junctionTable` here if you delete the variable assignment on my other comment

**Comment by closingin** [comment]
@GuillaumeCisco Change your PR type to "refactor", this is not a fix anymore (also amend your commits if possible) 🙂

File: services/analyzer/sequelize-tables-analyzer.js (hunk 1 above)
Comment (GuillaumeCisco) [reply services/analyzer/sequelize-tables-analyzer.js:142]: 👍

File: test-expected/sequelize/db-analysis-output/users.expected.json (hunk 2 above)
Comment (GuillaumeCisco) [reply test-expected/sequelize/db-analysis-output/users.expected.json:32]: 👍

**Comment by GuillaumeCisco** [comment]
> @GuillaumeCisco Change your PR type to "refactor", this is not a fix anymore (also amend your commits if possible) 🙂

Thank you! It is done 🚀

**Pull request merged**